pkg/prometheus/operator.go

```diff
@@ -308,19 +336,18 @@ func (c *Operator) syncNodeEndpoints() {
 
 	cache.ListAll(c.nodeInf.GetStore(), labels.Everything(), func(obj interface{}) {
 		n := obj.(*v1.Node)
-		for _, a := range n.Status.Addresses {
-			if a.Type == v1.NodeInternalIP {
-				eps.Subsets[0].Addresses = append(eps.Subsets[0].Addresses, v1.EndpointAddress{
-					IP:       a.Address,
-					NodeName: &n.Name,
-					TargetRef: &v1.ObjectReference{
-						Kind:       "Node",
-						Name:       n.Name,
-						UID:        n.UID,
-						APIVersion: n.APIVersion,
-					},
-				})
-			}
+		address, _, err := nodeAddress(n)
+		if err == nil {
```

Comment:
We should probably log that error and flip the condition to `err != nil` to move the larger block below one level to the left.